libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/internal/kapt/Kapt3KotlinGradleSubplugin.kt

```diff
@@ -386,6 +386,7 @@ class Kapt3GradleSubplugin @Inject internal constructor(private val registry: To
 
             kotlinCompilation.compileKotlinTaskProvider.configure {
                 (it as AbstractKotlinCompileTool<*>).setSource(sourcesOutputDir, kotlinSourcesOutputDir)
+                it.libraries.from(classesOutputDir)
```

Comment:
Not sure if this is the best way to add the directory but it does seem to work.

Reply:
You can use `                it.libraries.from(taskProvider.flatMap { kaptTask -> kaptTask.classesDir })` so that it captures the task dependency with classes output.

Reply:
Good call, should I change the line above too?